ADD SQL Objects mechanism to RDBMS commons (#51)
GoogleCloudPlatform/datacatalog-connectors-rdbms: google-datacatalog-rdbms-connector/README.md

To enable it you need 3 configuration files:
1. `ingest_cfg.yaml`
    ```yaml
    sql_objects:
      - name: 'functions'
        enabled: True
      - name: 'stored_procedures'
        enabled: True
    ```

    The `ingest_cfg.yaml` file must be located at the connection execution directory. You can find a sample in the [SAP HANA connector](../google-datacatalog-saphanaconnector/src/google/datacatalog_connectors/saphana/ingest_cfg.yaml).
    
    You can specify a list of sql objects with a flag to enable/disable it,
    it will be only considered if the `enabled` flag is set to `True`.
    
    The `query` and `metadata_definition` files use the 
    `sql_objects.name` value to be located, in the sample above,
     for the `function` SQL Object two files are expected: 
     `query_functions_sql_object.sql` and 
    `metadata_definition_functions_sql_object.sql`.
    
    And the `stored_procedures` SQL Objects expects:
    `query_stored_procedures_sql_object.sql` and 
    `metadata_definition_stored_procedures_sql_object.sql`.
    

1.  [query_functions_sql_object.sql](tests/google/datacatalog_connectors/rdbms/test_data/query_functions_sql_object.sql)  

>     Implement a query file that matches the sql object name, with the following pattern:

Reviewer: I believe it would be better in terms of readability if all **sql object** expressions were replaced with **SQL Object**. WDYT?

You can see working examples of implementing optional queries in the connectors code for PostgreSQL and MySQL. 
Please also refer to these samples to see how user configuration file `ingest_cfg.yaml` should look like.

### 4.2 Add support for SQL Objects
You can use the SQL Objects mechanism to ingest SQL objects like Functions, Stored Procedures,
Views, Materialized Views... from the SQL connector. This mechanism uses naming convention to locate
a SQL query and a metadata definition file.

You can see working examples of implementing SQL Objects in the test sources. 
To enable it you need 3 configurations:

> 1. [ingest_cfg.yaml](tests/google/datacatalog_connectors/rdbms/test_data/sql_objects_ingest_cfg.yaml)

Reviewer: The link is displayed as **ingest_cfg.yaml** but points to `tests/google/datacatalog_connectors/rdbms/test_data/sql_objects_ingest_cfg.yaml`. And there is a `ingest_cfg.yaml` file at `tests/google/datacatalog_connectors/rdbms/test_data`. It is a bit confusing... What are the relationships and expected content for these files?

Author: On the next PR I will update this to point to a sample implementation on the SAP HANA connector, while there are no connectors using this, I pointed out the test_data.

    | ---    | ---                                  |                    
    | key    | Key used to lookup the config files  |
    | type   | SQL Object type                      | 
    | name   | Name of the SQL Object               |
    | fields | Fields definition scraped from query |
    
    The `fields` attributes needs to implement the `source` and `target` definitions:
    
    | Field                      | Description                                                   |                     
    | ---                        | ---                                                           |                    
    | fields.source              | Name of the field scraped by the query                        |
    | fields.target              | Object with target field attributes                           | 
    | fields.target.field_name   | Name of the field when translated to Data Catalog attributes  |
    | fields.target.model        | Type of Data Catalog model, can be (tag or entry)             |
    | fields.target.type         | Type of the field to be translated to Data Catalog attributes |

    

> If any of those 3 configurations is not implemented, the SQL Object will be skipped and

Reviewer: ```suggestion
**The SQL Object will be skipped if any of the 3 configuration files are missing. **
```
A suggestion to highlight the behavior and keep it simple.

> won't be ingested.

Reviewer: ```suggestion

```
